src/__test__/components/data-exploration/heatmap/HeatmapPlot.test.jsx

```diff
@@ -1,545 +1,163 @@
-          },
-        },
-      },
-    });
-
-    component = mount(
-      <Provider store={store}>
-        <HeatmapPlot experimentId={experimentId} width={200} height={200} />
-      </Provider>,
+  it('Shows loader message if cellSets are loading', async () => {
+    const mockAPIResponses = _.merge(
+      generateDefaultMockAPIResponses(experimentId, fake.PROJECT_ID),
+      { [`experiments/${experimentId}/cellSets`]: () => stalledResponse },
     );
 
-    expect(component.find('HeatmapPlot').length).toEqual(1);
+    fetchMock.mockIf(/.*/, mockAPI(mockAPIResponses));
 
-    const expectedMarkerGenes = 5;
+    await loadAndRenderDefaultHeatmap();
 
-    expect(loadMarkerGenes).toHaveBeenCalledWith('123', 10, 'interactiveHeatmap', expectedMarkerGenes);
+    expect(screen.getByText(/We're getting your data.../i)).toBeInTheDocument();
   });
 
-  it('loads marker genes with a different amount of marker genes if louvain clusters are many', async () => {
-    const store = mockStore({
-      ...initialState,
-      networkResources: { environment: Environment.DEVELOPMENT },
-      cellSets: {
-        ...initialState.cellSets,
-        loading: true,
-        error: false,
-      },
-      genes: {
-        ...initialState.genes,
-        expression: {
-          ...initialState.genes.expression,
-          loading: false,
-          error: false,
-        },
-      },
-      experimentSettings: {
-        ...initialState.experimentSettings,
-        processing: {
-          ...initialState.processing,
-          configureEmbedding: {
-            clusteringSettings: { methodSettings: { louvain: { resolution: 10 } } },
-          },
-        },
-      },
-    });
+  it('Shows loader message if the marker genes are loading', async () => {
+    fetchMock.mockIf(/.*/, mockAPI(generateDefaultMockAPIResponses(experimentId, fake.PROJECT_ID)));
 
-    const louvainClusters = _.range(60).map((clusterIndex) => ({ key: `louvain-${clusterIndex}` }));
-    getCellSetsHierarchyByKeys.mockReturnValue(() => (
-      [{
-        key: 'louvain',
-        name: 'louvain clusters',
-        type: 'cellSets',
-        children: louvainClusters,
-      }]));
-
-    component = mount(
-      <Provider store={store}>
-        <HeatmapPlot experimentId={experimentId} width={200} height={200} />
-      </Provider>,
+    const customWorkerResponses = _.merge(
+      _.cloneDeep(mockWorkerResponses),
+      { '5-marker-genes': () => stalledResponse },
     );
 
-    expect(component.find('HeatmapPlot').length).toEqual(1);
+    seekFromAPI.mockClear();
+    seekFromAPI.mockImplementation((a, b, c, requested) => customWorkerResponses[requested]());
 
-    const expectedMarkerGenes = 3;
+    await loadAndRenderDefaultHeatmap();
 
-    expect(loadMarkerGenes).toHaveBeenCalledWith('123', 10, 'interactiveHeatmap', expectedMarkerGenes);
+    expect(screen.getByText(/We're getting your data.../i)).toBeInTheDocument();
```

Comment:
I am very pedantic here, but can you add space between `data` and `...` in here and in the actual code implementation text?

Reply:
there should be spaces between these